CalibTracker/SiStripChannelGain/plugins/SiStripGainFromAsciiFile.cc

```diff
@@ -60,39 +71,36 @@ SiStripApvGain * SiStripGainFromAsciiFile::getNewObject(){
   ss << "[SiStripGainFromAsciiFile::getNewObject]\n Filling SiStripApvGain object";
   short nApvPair;
   for(std::vector<uint32_t>::const_iterator it=DetIds.begin(); it!=DetIds.end(); it++){
-
+    ModuleGain MG;
     if (DetId(*it).det()!=DetId::Tracker)
       continue;
 
     nApvPair=reader.getNumberOfApvsAndStripLength(*it).first/2;
     
     ss << "Looking at detid " << *it << " nApvPairs  " << nApvPair << std::endl;
-
-    __gnu_cxx::hash_map< unsigned int,FibersGain>::const_iterator iter=GainsMap.find(*it);
-    
+    __gnu_cxx::hash_map< unsigned int,ModuleGain>::const_iterator iter=GainsMap.find(*it);    
     if (iter!=GainsMap.end()){
-      FG = iter->second;
-      ss << " " <<  FG.fiber[0] << " " <<  FG.fiber[1] << " " <<  FG.fiber[2] << std::endl;
-    }
-    else {
+      MG = iter->second;
+      ss << " " <<  MG.apv[0] << " " <<  MG.apv[1] << " " <<  MG.apv[2] <<  " " <<  MG.apv[3] << " " <<  MG.apv[4] << " " <<  MG.apv[5] << std::endl;
+    }else {
       ss << "Hard reset for detid " << *it << std::endl;
-      FG.hard_reset(referenceValue_);
+      MG.hard_reset(referenceValue_);
     }
     
     std::vector<float> DetGainsVector;
     
     if (nApvPair==2){
-      DetGainsVector.push_back(FG.fiber[0]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[0]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[2]/referenceValue_);        
-      DetGainsVector.push_back(FG.fiber[2]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[0]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[1]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[2]/referenceValue_);        
+      DetGainsVector.push_back(MG.apv[3]/referenceValue_);  
     } else if (nApvPair==3){   		   		   		  
-      DetGainsVector.push_back(FG.fiber[0]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[0]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[1]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[1]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[2]/referenceValue_);  
-      DetGainsVector.push_back(FG.fiber[2]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[0]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[1]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[2]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[3]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[4]/referenceValue_);  
+      DetGainsVector.push_back(MG.apv[5]/referenceValue_);  
```

Comment:
@quertenmont 

this can all be replaced by 
if ( nAprPair==2 || nApvPair==3) for (unsigned int i=0;i<2*nApvPair; i++) DetGainsVector.push_back(MG.apv[i]/referenceValue_); 

right? (ok, would be better also to cache 1.0/referenceValue_ and use * instead of /)